nixos/modules/services/games/minecraft-server.nix

```diff
@@ -22,6 +22,15 @@ let
   '' + concatStringsSep "\n" (mapAttrsToList
     (n: v: "${n}=${cfgToString v}") cfg.serverProperties));
 
+  stopScript = pkgs.writeScript "minecraft-stop" ''
+    #!${pkgs.runtimeShell}
+
+    echo stop > ${config.systemd.sockets.minecraft-server.socketConfig.ListenFIFO}
+
+    while kill -0 "$1" 2> /dev/null; do
+      sleep 1s
+    done
+  '';
```

Comment:
`writeShellScriptBin` already adds the shebang:

```suggestion
  stopScript = pkgs.writeShellScriptBin "minecraft-stop" ''
    echo stop > ${config.systemd.sockets.minecraft-server.socketConfig.ListenFIFO}

    while kill -0 "$1" 2> /dev/null; do
      sleep 1s
    done
  '';
```

Reply:
I swapped it to `writeShellScript`, but i did not use the `Bin` version as this is not a binary meant to be run by a user, only to be called directly by systemd inside the `ExecStop` part of the minecraft-server.service file. This use should make it clear that it is not a binary you should ever add to your PATH.

Reply:
You're right, not sure why I thought `writeShellScriptBin` was the right one, as opposed to `writeShellScript`. (I was thinking of the behavior of `writeShellScript`, so no clue what happened there.)